### PR TITLE
Double the number of processor build-and-test targets

### PR DESCRIPTION
**Pull request by jmacd**
Description: Double the number of processor test groups so that new tests can be added, e.g., the new tests in
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31946

Testing: These changes were applied in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31946 and they appeared to help with spurious test timeouts.

Similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32496

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,9 @@ RECEIVER_MODS_1 := $(shell find ./receiver/[g-o]* $(FIND_MOD_ARGS) -exec $(TO_MO
 RECEIVER_MODS_2 := $(shell find ./receiver/[p]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) ) # Prometheus is special and gets its own section.
 RECEIVER_MODS_3 := $(shell find ./receiver/[q-z]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
 RECEIVER_MODS := $(RECEIVER_MODS_0) $(RECEIVER_MODS_1) $(RECEIVER_MODS_2) $(RECEIVER_MODS_3)
-PROCESSOR_MODS := $(shell find ./processor/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+PROCESSOR_MODS_0 := $(shell find ./processor/[a-o]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+PROCESSOR_MODS_1 := $(shell find ./processor/[p-z]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+PROCESSOR_MODS := $(PROCESSOR_MODS_0) $(PROCESSOR_MODS_1)
```
Comment (jmacd) [inline Makefile:33]: This gives an 12/11 breakdown.

**Review by songy23 — APPROVED** [review]

**Comment by songy23** [comment]
Need a rebase

**Comment by jmacd** [comment]
This will need a rebase again following https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32496, which is ready to merge.

**Review by crobert-1 — APPROVED** [review]
We probably want to wait on this until https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32496 is merged.

**Review by codeboten — APPROVED** [review]

**Pull request merged**